tools/dependency/release_dates.py

```diff
@@ -63,31 +62,53 @@ def verify_and_print_latest_release(dep, repo, metadata_version, release_date, c
         # check for --check_deps flag, To run this only on github action schedule
         # and it does not bloat CI on every push
         if create_issue:
-            create_issues(dep, metadata_version, release_date, latest_release)
+            create_issues(dep, repo, metadata_version, release_date, latest_release)
+
+
+def is_sha(text):
+    if len(text) != 40:
+        return False
+    try:
+        int(text, 16)
+    except ValueError:
+        return False
+    return True
 
 
 # create issue for stale dependency
-def create_issues(dep, metadata_version, release_date, latest_release):
+def create_issues(dep, package_repo, metadata_version, release_date, latest_release):
     """Create issues in GitHub.
 
     Args:
         dep : name of the deps
-        metadata_version :
+        package_repo: package Url
+        metadata_version: current version information
         release_date : old release_date
         latest_release : latest_release (name and date )
     """
     access_token = os.getenv('GITHUB_TOKEN')
     git = github.Github(access_token)
-    repo = git.get_repo('envoyproxy/envoy')
+    repo = git.get_repo(GITHUB_REPO_LOCATION)
     # Find GitHub label objects for LABELS.
     labels = []
     for label in repo.get_labels():
         if label.name in LABELS:
             labels.append(label.name)
     if len(labels) != len(LABELS):
         raise DependencyUpdateError('Unknown labels (expected %s, got %s)' % (LABELS, labels))
-    body = f'*WARNING* {dep} has a newer release than {metadata_version}@<{release_date}>:{latest_release.tag_name}@<{latest_release.created_at}>'
-    title = f'Newer release available {dep}: {latest_release.tag_name}'
+    # trunctate metadata_version to 7 char if its sha_hash
+    if is_sha(metadata_version):
+        metadata_version = metadata_version[0:7]
+        # dedent for making multiline string works in github specailly link
+    body = f"""\
```

Comment:
if you move these to constants as templates i think it would improve the code - it still needs not to be indented - but using a template should be quite straightforward

doing so will move this out of the logic which i think is preferable

Reply:
eg

```python
BODY_TPL = """
Package Name: ${dep}
Current Version: ${metadata_version}@${release_date}
Available Version: ${latest_release.tag_name}@${latest_release.created_at}
Upstream link: https://github.com/${package_repo.full_name}
"""
```

Reply:
which you can use as 

```python
string.Template(BODY_TPL).substitute(dep=..., ...)

```

Reply:
yea that' why did. but it also generating some tabs in the starting due to which Github was taking it as code blocks. 
Let me try this again tho.

Reply:
if you set the template as constant with no indent, it will have no indent

Reply:
hmm Got it. Have to mv `BODY_TPL` out as a Global var.